caddyhttp/httpserver/replacer.go

```diff
@@ -232,15 +232,23 @@ func (r *replacer) getSubstitution(key string) string {
 	case "{path}":
 		// if a rewrite has happened, the original URI should be used as the path
 		// rather than the rewritten URI
-		path := r.request.Header.Get("Caddy-Rewrite-Original-URI")
-		if path == "" {
+		var path string
+		origpath := r.request.Header.Get("Caddy-Rewrite-Original-URI")
+		if origpath == "" {
 			path = r.request.URL.Path
+		} else {
+			origURL, _ := url.Parse(origpath)
+			path = origURL.Path
 		}
 		return path
 	case "{path_escaped}":
-		path := r.request.Header.Get("Caddy-Rewrite-Original-URI")
-		if path == "" {
+		var path string
+		origpath := r.request.Header.Get("Caddy-Rewrite-Original-URI")
```

Comment:
This reminds me, we're switching to using the `context` package... #1331 does this for something similar, and frankly I only used a header field because context didn't exist at the time so it was kind of a hack. What do you think if we switch this to use req.Context? (It can be a separate PR, I prefer multiple smaller PRs!)